### ISSUE TITLE
Allow for dynamic field/segment occurrences based on another field in the same record

### ISSUE THREAD

**Issue by GoogleCodeExporter** [unlockha/beanio#69]
```
See summary.
```

Original issue reported on code.google.com by `kevin.s...@gmail.com` on 6 Apr 2013 at 6:23

**Comment by GoogleCodeExporter** [comment]
```
A new configuration attribute called 'occursRef' may be applied to a field or 
segment that refers to the name of a preceding field that indicates the number 
of occurrences.  For example:

  <stream name="o1" format="csv">
    <record name="record" class="map">
      <field name="occurs" ignore="true" />
      <field name="values" collection="list" occursRef="occurs" />
      <field name="end" literal="end" />
    </record>
  </stream>

Would allow for unmarshaling the following records:

0,end
1,one,end
2,one,two,end
etc...

```

Original comment by `kevin.s...@gmail.com` on 8 Apr 2013 at 1:56
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
Resolved for 2.1.0.M1.  Snapshot JAR attached.
```

Original comment by `kevin.s...@gmail.com` on 8 Apr 2013 at 2:25

Attachments:
- [beanio.jar](https://storage.googleapis.com/google-code-attachments/beanio/issue-69/comment-2/beanio.jar)

**Comment by GoogleCodeExporter** [comment]
Original comment by `kevin.s...@gmail.com` on 27 Apr 2013 at 11:44
- Changed state: **Fixed**
